### DISCUSSION TITLE
Why is the ‘BackColor’ in the ‘glyph index’ protocol actually ‘ForeColor’, and ‘ForeColor’ actually ‘BackColor’?

### DISCUSSION THREAD

**Discussion by Mr-LZG** [FreeRDP/FreeRDP#10404] [General]
Why is the ‘BackColor’ in the ‘glyph index’ protocol actually ‘ForeColor’, and ‘ForeColor’ actually ‘BackColor’?

**Comment by akallabeth** [comment]
@Mr-LZG please read https://github.com/FreeRDP/FreeRDP/wiki/BugReporting that has a lot of hints on how to create issues (and discussions for that matter)
try to be a bit more specific with details and link to what you mean for example.

also: [MS-RDPBCGR](https://learn.microsoft.com/en-us/openspecs/windows_protocols/MS-RDPBCGR/5073f4ed-1e93-45e1-b039-6e30c385867c) contains protocol details.

**Reply by Mr-LZG** [reply]
Thank you for your prompt response, and I apologize for submitting a question that made you feel bad. And I read the MS-RDPBCGR and MS-RDPEGDI documents and did not find the answer I was looking for.

**Reply by akallabeth** [reply]
does not help in identifying what you mean ;)
which functions are you talking about?

**Reply by Mr-LZG** [reply]
I found that the "BackColor" in the "glyph index" protocol actually corresponds to the color of the text on the RDP server; and the "ForeColor" is actually the background color of the text on the RDP server.   I don't know why this is.

**Reply by akallabeth** [reply]
link please.

**Reply by Mr-LZG** [reply]
![image](https://github.com/user-attachments/assets/6ac80264-a8f1-400e-9f48-49e865ac5b73)

There is no link, but you can refer to the source code of xrdp

**Reply by Mr-LZG** [reply]
In xrdp, bgcolor is assigned to fgColor, and fgColor is assigned to bgColor.

**Reply by akallabeth** [reply]
you are talking about your code?
well, then you should know ;)

for the rest there is [MS-RDPEGDI](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpegdi/8e906465-a1da-4d06-91db-54beb7827381)

**Reply by Mr-LZG** [reply]
I am not discussing my code, but I found this special place when migrating from xrdp to freerdp. I did not find relevant instructions in the MS-RDPEGDI protocol, so I would like to ask you for help.

**Reply by akallabeth** [reply]
read the link, there all the parameters of the `GLYPHINDEX_ORDER` are documented there.

**Reply by Mr-LZG** [reply]
Thank you very much. I didn't read the document carefully enough. I just read it again and understood it.
